artiq/coredevice/urukul.py

```diff
@@ -198,8 +201,11 @@ def init(self, blind=False):
             self.cfg_write(self.cfg_reg)
         else:
             proto_rev = urukul_sta_proto_rev(self.sta_read())
-            if proto_rev != STA_PROTO_REV_MATCH:
+            if proto_rev not in STA_PROTO_REV_MATCH:
                 raise ValueError("Urukul proto_rev mismatch")
+            if bool(cfg & CFG_CLK_SEL1) \
+               and proto_rev == STA_PROTO_REV_MATCH[0]:
+                raise ValueError("Unsupported clocking option")
```

Comment:
That's the question. Is it worth trying to unify the cpld gateware over v1.2 and v1.3, not icrement the protocol and ignore the missing functionality on v1.2 **or** should we be strict and add the complexity of supporting multiple protocol versions and different gateware for different hardware.
I am leaning towards the latter (and documenting the NOP-ness of clk_sel1 on v1.2/earlier).

Reply:
> I am leaning towards the latter (and documenting the NOP-ness of clk_sel1 on v1.2/earlier).

"latter" or "former"?

> I am leaning towards the latter (and documenting the NOP-ness of clk_sel1 on v1.2/earlier).

Where do you want this documented? Just the constructor?

Reply:
OK, let's do that.

I'll update the PR later today